cmd/node/node.go

```diff
@@ -554,11 +550,16 @@ func (app *SpacemeshApp) initServices(ctx context.Context,
 	atxdb := activation.NewDB(atxdbstore, idStore, mdb, layersPerEpoch, goldenATXID, validator, app.addLogger(AtxDbLogger, lg))
 	tBeaconDB := tortoisebeacon.NewDB(tBeaconDBStore, app.addLogger(TBeaconDbLogger, lg))
 
-	// TODO(nkryuchkov): Enable weak coin when finished.
-	// wc := weakcoin.NewWeakCoin(weakcoin.DefaultThreshold(), nodeID, swarm, signing.VRFVerify, vrfSigner, app.addLogger(WeakCoinLogger, lg))
-	wc := weakcoin.ValueMock{Value: false}
+	wcConfig := weakcoin.DefaultConfig()
+	wcConfig.MaxRound = types.RoundID(app.Config.TortoiseBeacon.RoundsNumber)
+	wc := weakcoin.New(swarm,
+		vrfSigner, signing.VRFVerifier{},
```

Comment:
I'd add a constructor for `signing.VRFVerifier`.

```suggestion
		vrfSigner, signing.NewVRFVerifier(),
```

Reply:
but this is already a valid initializer. why do you want to have a function?

Reply:
I just prefer initializing via a constructor function if some struct with methods is created. But it's a matter of taste